sources/packages/backend/libs/integrations/src/cas/cas.service.ts

```diff
@@ -267,6 +246,28 @@ export class CASService {
     return supplierAddress;
   }
 
+  /**
+   * Handles bad request errors from CAS API.
+   * @param error The error object caught during the request.
+   * @param defaultMessage The default message to throw for non-specific errors.
+   */
+  private handleBadRequestError(error: unknown, defaultMessage: string): never {
+    if (
+      error instanceof AxiosError &&
+      error.response?.status === HttpStatus.BAD_REQUEST &&
+      !!error.response?.data[CAS_RETURNED_MESSAGES]
+    ) {
+      const casKnownErrors =
+        error.response.data[CAS_RETURNED_MESSAGES].toString();
+      throw new CustomNamedError(
+        "CAS Bad Request Errors",
+        CAS_BAD_REQUEST,
+        casKnownErrors.split(" | "),
```

Comment:
I would recommend executing the split by the "|" and then doing a `trim()` to remove spaces, if any.

Reply:
The recommendation was to do the split by "|" not " | ".
We can review it when we add the unit test as below comment
https://github.com/bcgov/SIMS/pull/4017#discussion_r1872276210